ibis/config.py

```diff
@@ -200,7 +202,17 @@ def __getattr__(self, key):
         try:
             v = self.d[key]
         except KeyError as e:
-            raise AttributeError(*e.args)
+            # See if the options doesn't exist because it's from a backend
+            # not yet loaded. Try to load, and try the option again.
+            try:
```

Comment:
do we go down this path in tests?

Reply:
Good point. Looks like we don't have any tests for options. Our options code is a copy of the pandas one, which does have tests. I don't think copying pandas tests, or implementing new ones are great ideas. If we want to do things properly, we should create a new package for the options code, and use it in both projects.

Let me know your thoughts, but in any case, I don't think this is for this PR.

Reply:
i think you can remove *all* of this because you *already* load the options in the backend loading itself. I woudn't load the backend here at all.

Reply:
Unfortunately this is needed. See this code:
```python
ibis.options.impala. temp_table = 'foo'
ibis.impala.connect(...)
```
Without this code, the first line will fail, since there is no `impala.temp_table` option registered. The loading of the backend you're referring to is loaded in the second line.

Reply:
this is easy to replace by
```
import ibis.impala
ibis.options.impala.....
ibis.impala.connect()
```
right?

Reply:
It would be:
```python
import ibis
ibis.impala
ibis.options.impala.temp_table = 'foo'
ibis.impala.connect()
```
But, yes, that's correct, if we are happy that the next code fails when setting the option, this is not needed:
```python
import ibis
ibis.options.impala.temp_table = 'foo'
ibis.impala.connect()
```

Reply:
see comments above

Reply:
i don't really love all of these nested try/excepts